lua/ttt2/extensions/draw.lua

```diff
@@ -0,0 +1,436 @@
+---
+-- draw extension functions
+-- @author Mineotopia, LeBroomer, Alf21, saibotk
+
+AddCSLuaFile()
+
+if SERVER then return end
+
+local render = render
+local surface = surface
+local draw = draw
+local drawSimpleText = draw.SimpleText
+local table = table
+local cam = cam
+local tableCopy = table.Copy
+local mathRound = math.Round
+
+local colorShadowDark = Color(0, 0, 0, 220)
+local colorShadowBright = Color(0, 0, 0, 75)
+
+local function GetShadowColor(color)
+	local tmpCol = color.r + color.g + color.b > 200 and tableCopy(colorShadowDark) or tableCopy(colorShadowBright)
+
+	tmpCol.a = mathRound(tmpCol.a * (color.a / 255))
+
+	return tmpCol
+end
+
+---
+-- A function to draw an outlined box with a definable width
+-- @param number x The x position of the rectangle
+-- @param number y The y position of the rectangle
+-- @param number w The width of the rectangle
+-- @param number h The height of the rectangle
+-- @param [default=1] number t The thickness of the line
+-- @param [default=Color(255,255,255,255)] Color color The color of the line
+-- @2D
+-- @realm client
+function draw.OutlinedBox(x, y, w, h, t, color)
+	t = t or 1
+
+	surface.SetDrawColor(color or COLOR_WHITE)
+
+	for i = 0, t - 1 do
+		surface.DrawOutlinedRect(x + i, y + i, w - i * 2, h - i * 2)
+	end
+end
+
+local drawOutlinedBox = draw.OutlinedBox
+
+---
+-- A function to draw an outlined box with a shadow
+-- @param number x The x position of the rectangle
+-- @param number y The y position of the rectangle
+-- @param number w The width of the rectangle
+-- @param number h The height of the rectangle
+-- @param [default=1]number t The thickness of the line
+-- @param [default=Color(255,255,255,255)]Color color The color of the line
+-- @param [default=1.0]number scale A scaling factor that is used for the shadows
+-- @2D
+-- @realm client
+function draw.OutlinedShadowedBox(x, y, w, h, t, color, scale)
+	color = color or COLOR_WHITE
+	scale = scale or 1
+
+	local shift1 = mathRound(scale)
+	local shift2 = mathRound(scale * 2)
+
+	local tmpCol = GetShadowColor(color)
+
+	drawOutlinedBox(x + shift2, y + shift2, w, h, t, tmpCol)
+	drawOutlinedBox(x + shift1, y + shift1, w, h, t, tmpCol)
+	drawOutlinedBox(x + shift1, y + shift1, w, h, t, tmpCol)
+	drawOutlinedBox(x, y, w, h, t, color)
+end
+
+---
+-- A function to draws a simple box without a radius
```

Comment:
why is it needed to specify "without a radius" what is the "radius" you speak of?